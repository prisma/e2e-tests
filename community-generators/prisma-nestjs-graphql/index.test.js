const fs = require('fs')

describe('test generator', () => {
  it('should produce same prisma ts file for same Prisma schema', () => {
    const content = fs.readFileSync('./prisma-nestjs-graphql/prisma/affected-rows.output.ts', {
      encoding: 'utf8',
    })
    expect(content).toMatchSnapshot()
  })

  it('should produce same user ts file for same Prisma schema', () => {
    const content = fs.readFileSync('./prisma-nestjs-graphql/user/aggregate-user.args.ts', {
      encoding: 'utf8',
    })
    expect(content).toMatchSnapshot()
  })
})
