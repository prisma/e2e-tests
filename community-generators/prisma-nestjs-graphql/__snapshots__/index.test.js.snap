// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test generator should produce same prisma ts file for same Prisma schema 1`] = `
"import { Field } from '@nestjs/graphql';
import { ObjectType } from '@nestjs/graphql';
import { Int } from '@nestjs/graphql';

@ObjectType()
export class AffectedRows {
    @Field(() => Int, {nullable:false})
    count!: number;
}
"
`;

exports[`test generator should produce same user ts file for same Prisma schema 1`] = `
"import { Field } from '@nestjs/graphql';
import { ArgsType } from '@nestjs/graphql';
import { UserWhereInput } from './user-where.input';
import { UserOrderByInput } from './user-order-by.input';
import { UserWhereUniqueInput } from './user-where-unique.input';
import { Int } from '@nestjs/graphql';
import { UserCountAggregateInput } from './user-count-aggregate.input';
import { UserMinAggregateInput } from './user-min-aggregate.input';
import { UserMaxAggregateInput } from './user-max-aggregate.input';

@ArgsType()
export class AggregateUserArgs {
    @Field(() => UserWhereInput, {nullable:true})
    where?: UserWhereInput;

    @Field(() => [UserOrderByInput], {nullable:true})
    orderBy?: Array<UserOrderByInput>;

    @Field(() => UserWhereUniqueInput, {nullable:true})
    cursor?: UserWhereUniqueInput;

    @Field(() => Int, {nullable:true})
    take?: number;

    @Field(() => Int, {nullable:true})
    skip?: number;

    @Field(() => UserCountAggregateInput, {nullable:true})
    count?: UserCountAggregateInput;

    @Field(() => UserMinAggregateInput, {nullable:true})
    min?: UserMinAggregateInput;

    @Field(() => UserMaxAggregateInput, {nullable:true})
    max?: UserMaxAggregateInput;
}
"
`;
